components/tracker/src/lib.rs

```diff
@@ -147,3 +153,36 @@ pub struct RequestMetrics {
     pub apply_write_wal_nanos: u64,
     pub apply_write_memtable_nanos: u64,
 }
+
+pub struct TrackGuard<M>
+where
+    M: FnMut(&mut RequestMetrics) -> &mut u64,
+{
+    token: TrackerToken,
+    field: M,
+    pub value: u64,
+}
+
+impl<M> TrackGuard<M>
+where
+    M: FnMut(&mut RequestMetrics) -> &mut u64,
+{
+    pub fn new(token: TrackerToken, field: M) -> TrackGuard<M> {
+        TrackGuard {
+            token,
+            value: 0,
+            field,
+        }
+    }
+}
+
+impl<M> Drop for TrackGuard<M>
+where
+    M: FnMut(&mut RequestMetrics) -> &mut u64,
+{
+    fn drop(&mut self) {
```

Comment:
We can do this only for values that are not 0, reducing the lock acquisition overhead for metrics that are often 0, such as TOTAL_THROTTLE_DURATION

Reply:
Done